From 28078a1dc1a29210061460ea479ba5b31a23702a Mon Sep 17 00:00:00 2001
From: rajucm <raju.mallikarjun.chegaraddi@intel.com>
Date: Mon, 27 Mar 2023 17:50:31 +0000
Subject: [PATCH] fix for static analysis issues

i915 module: avoid double free
btintel module: fix kmalloc validation by using right variable

Tracked-On: OAM-105549
Signed-off-by: rajucm <raju.mallikarjun.chegaraddi@intel.com>
---
 drivers/bluetooth/btintel.c     | 2 +-
 drivers/gpu/drm/i915/i915_gem.c | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/bluetooth/btintel.c b/drivers/bluetooth/btintel.c
index d2809e4fbe05..322468a2c98f 100644
--- a/drivers/bluetooth/btintel.c
+++ b/drivers/bluetooth/btintel.c
@@ -2365,7 +2365,7 @@ static int btintel_get_codec_config_data(struct hci_dev *hdev,
 	}
 
 	*ven_data = kmalloc(sizeof(__u8), GFP_KERNEL);
-	if (!*ven_data) {
+	if (*ven_data == NULL) {
 		err = -ENOMEM;
 		goto error;
 	}
diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 58e00a3762f4..519ef337f1a6 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -2167,9 +2167,7 @@ int i915_gem_open(struct drm_i915_private *i915, struct drm_file *file)
 	kfree(file_priv->process_name);
 out_free_file:
 	put_pid(file_priv->tgid);
-	kfree(file_priv);
 #endif
-
 err_context:
 	i915_drm_client_put(client);
 err_client:
-- 
2.34.1

